logstash-core/lib/logstash/pipeline.rb

```diff
@@ -34,6 +34,7 @@ module LogStash; class BasePipeline
 
   def initialize(config_str, settings = SETTINGS)
     @logger = self.logger
+    @mutex = Mutex.new
```

Comment:
hmmm ... unless I am misunderstanding I'd really advise against using a "general purpose" mutex ... a mutex should have a single purpose IMO ... if you start mixing thread safety concerns by "sharing" a single mutex it will hard to understand and hard de debug.